src/rpc/blockchain.cpp

```diff
@@ -1532,6 +1539,262 @@ UniValue getchaintxstats(const JSONRPCRequest& request)
+        }
+    }
+}
+
+UniValue getblockstats(const JSONRPCRequest& request)
+{
+    std::set<std::string> valid_plot_values = {
+        "height",
+        "time",
+        "mediantime",
+        "txs",
+        "ins",
+        "outs",
+        "subsidy",
+        "totalfee",
+        "reward",
+        "utxo_increase",
+        "utxo_size_inc",
+        "total_size",
+        "total_weight",
+        "total_out",
+        "minfee",
+        "maxfee",
+        "medianfee",
+        "medianfeerate",
+        "medianfeerate_old",
+        "minfeerate",
+        "minfeerate_old",
+        "maxfeerate",
+        "maxfeerate_old",
+        "avgfeerate",
+        "avgfeerate_old",
+        "avgfee",
+    };
+    if (request.fHelp || request.params.size() < 1 || request.params.size() > 4)
+        throw std::runtime_error(
+            "getblockstats ( nStart nEnd plotValues )\n"
+            "\nCompute per block statistics for a given window.\n"
+            "\nArguments:\n"
+            "1. \"start\"      (numeric, required) The height of the block that starts the window.\n"
+            "2. \"end\"        (numeric, optional) The height of the block that ends the window (default: current tip).\n"
+            "3. \"plotvalues\"  (string,  optional) Values to plot (comma separated), default(all): " + boost::join(valid_plot_values, ",") +
+            "\nResult:\n"
+            "{\n"
+            "TODO"
+            "}\n"
+            "\nExamples:\n"
+            + HelpExampleCli("getblockstats", "1000 1000 \"minfeerate,avgfeerate\"")
+            + HelpExampleRpc("getblockstats", "1000 1000 \"maxfeerate,avgfeerate\"")
+        );
+
+    LOCK(cs_main);
+
+    int start = request.params[0].get_int();
+    if (start < 1 || start > chainActive.Height()) {
+        throw JSONRPCError(RPC_INVALID_PARAMETER, "Start block height out of range");
+    }
+
+    int end;
+    if (request.params.size() > 1) {
+        end = request.params[1].get_int();
+        if (end < 0 || end > chainActive.Height()) {
+            throw JSONRPCError(RPC_INVALID_PARAMETER, "End block height out of range");
+        }
+    } else {
+        end = chainActive.Height();
+    }
+
+    std::set<std::string> plot_values;
+    if (request.params.size() > 2) {
+        boost::split(plot_values, request.params[2].get_str(), boost::is_any_of(","));
+
+        for (const std::string& plot_value : plot_values) {
+            if (valid_plot_values.count(plot_value) == 0) {
+                throw JSONRPCError(RPC_INVALID_PARAMETER, strprintf("Invalid plot value %s", plot_value));
+            }
+        }
+    } else {
+        plot_values = valid_plot_values;
+    }
+
+    std::map<std::string, UniValue> map_stats;
+    for (const std::string& plot_value : plot_values) {
+        map_stats[plot_value] = UniValue(UniValue::VARR);
+    }
+
+    CBlockIndex* pindex = chainActive[end];
+    for (int i = end; i >= start; i--) {
+        UpdateBlockStats(pindex, plot_values, map_stats);
```

Comment:
Maybe. I'll wait for someone else to weight in since it doesn't seem a big deal either way.